### PR TITLE
[ENH] torch based ResNet classifier and regressor

### LINKED ISSUE #9321: [ENH] Implement `torch` based resnet time series classifier and regressor
## Is your feature request related to a problem? Please describe.
This is a sub-issue of #8699 
Implement resnet time series classifier and regressor in `torch`, based on the old `tensorflow` implementations which can be found in `sktime/sktime/networks/`, `sktime/sktime/classification/deep_learning/` and `sktime/sktime/regressor/deep_learning/`

## Describe the solution you'd like
[SimpleRNNClassifierTorch](https://github.com/sktime/sktime/blob/main/sktime/classification/deep_learning/rnn/_rnn_torch.py) implemented in [PR/#8842](https://github.com/sktime/sktime/pull/8842) can be used as a template for classifier migrations. Similarly, [SimpleRNNRegressorTorch](https://github.com/sktime/sktime/blob/main/sktime/regression/deep_learning/rnn/_rnn_torch.py) implemented in [PR/#9013](https://github.com/sktime/sktime/pull/9013) can be used as a template for regressor migrations.
Discussion in comment section of both the above mentioned PRs gives an idea behind certain choices made and also possible improvements that can be made, should you need to refer.

Please include the following checklist in your PR.
- [ ] Implemented the network structure in `sktime/sktime/networks/`
- [ ] Implemented the classifier in `sktime/sktime/classification/deep_learning/`
- [ ] Implementded the regressor in `sktime/sktime/regressor/deep_learning/`
- [ ] Followed the naming convention and directory structure from RNN migration samples
- [ ] Followed the [sktime docstring convention](https://www.sktime.net/en/latest/developer_guide/documentation.html#docstring-conventions). Please refer the RNN migration samples to understand the usage.
- [ ] Added meaningful test parameters in the classmethod `get_test_params` for classifiers and regressors.
- [ ] Followed the [sktime guide on implementing estimators](https://www.sktime.net/en/latest/developer_guide/add_estimators.html)
- [ ] Used same defaults in `torch` implemention as used in the corresponding `tensorflow` implementation of same model.

## Additional context
[Issue/#8928](https://github.com/sktime/sktime/issues/8928) consolidates further improvements/enhancements that are desirable and will be appreciated if you choose to incorporate. Feel free to brainstorm or share your ideas on these.

### PR DESCRIPTION
**Pull request by DCchoudhury15**
#### Reference Issues/PRs
Fixes #9321

#### What does this implement/fix? Explain your changes.
Implements PyTorch based ResNet time series classifier and regressor, migrating the existing TensorFlow implementation to PyTorch. Follows the same pattern as the RNN migration (PRs #8842, #9013).

- Added `ResNetNetworkTorch` - core 3-block residual network backbone in PyTorch
- Added `ResNetClassifierTorch` -time series classifier using the ResNet backbone
- Added `ResNetRegressorTorch` - time series regressor using the ResNet backbone
- Refactored the existing `resnet.py` files into proper subpackages (`resnet/`) to accommodate both TensorFlow and PyTorch backends side by side
- Updated API reference documentation for both classification and regression

#### Does your contribution introduce a new dependency? If yes, which one?
No new dependencies. `torch` is already a soft dependency in sktime.

#### What should a reviewer concentrate their feedback on?
- Correctness of the 3-block residual network architecture in `sktime/networks/resnet/_resnet_torch.py` - specifically the shortcut connections and channel dimensions across blocks
- Consistency with the existing PyTorch base classes `BaseDeepClassifierPytorch` and `BaseDeepRegressorTorch`
- The refactoring of `resnet.py` into `resnet/` subpackages - ensuring the TF classes are not broken

#### Did you add any tests for the change?
No explicit new tests added. The estimators are registered with `get_test_params()` and will be automatically picked up by sktime's `TestAllClassifiers` and `TestAllRegressors` test suites.

#### Any other comments?
 This PR  is adding PyTorch support for the ResNet architecture, which was previously only available in TensorFlow. The implementation closely follows the existing RNN PyTorch migration as a reference template.Any reviews or changes are appreciated.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
- [x] Optionally, for added estimators: I've added myself to the `maintainers` tag.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG].

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`.
- [x] I've added one or more illustrative usage examples to the docstring.
- [x] The estimator relies on `torch` as a soft dependency -`python_dependencies` tag is set and dependency isolation is handled via `_safe_import`.

### CODE REVIEW

**Comment by DCchoudhury15** [comment]
Hi @RecreationalMath , apologies for the late pr, changes i made in this =>
Ported the ResNet classifier and regressor from TF to PyTorch, following the RNN migration pattern. Also refactored the resnet.py files into proper subpackages so both backends sit cleanly together.
A few things I'm not fully sure about and would love feedback on  the tensor shape handling in forward(), whether padding="same" could cause issues on older torch versions, whether random_state seeding is fully handled by the base class or needs to be explicitly seeded in the network too, and whether the activation_hidden validation in the network is the right place to do it or if the base class should handle that.
Happy to fix anything and looking forward to the review!

File: sktime/networks/resnet/_resnet_torch.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        self.activation_hidden = activation_hidden
+        self.random_state = random_state
+
+        super().__init__()
+
+        Conv1d = _safe_import("torch.nn.Conv1d")
+        BatchNorm1d = _safe_import("torch.nn.BatchNorm1d")
+        Linear = _safe_import("torch.nn.Linear")
+
+        # --- Block 1 ---
```
Comment (RecreationalMath) [inline sktime/networks/resnet/_resnet_torch.py]: Multiple blocks are defined, and values are hardcoded. These settings should not be hardcoded and instead inferred from the user input after exposing it to the user.

File: sktime/networks/resnet/_resnet_torch.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        self.conv1_x = Conv1d(
+            input_size, self.n_feature_maps, kernel_size=8, padding="same"
```
Comment (RecreationalMath) [inline sktime/networks/resnet/_resnet_torch.py]: `kernel_size`, `padding` these should be exposed and inferred from user inputs.

File: sktime/networks/resnet/_resnet_torch.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            self.n_feature_maps * 2,
+            self.n_feature_maps * 2,
+            kernel_size=3,
+            padding="same",
+        )
+        self.bn3_z = BatchNorm1d(self.n_feature_maps * 2)
+        self.shortcut3_bn = BatchNorm1d(self.n_feature_maps * 2)
+
+        # --- Final layers ---
+        _supported = {
```
Comment (RecreationalMath) [inline sktime/networks/resnet/_resnet_torch.py]: Not sure, why the activations are constrained to just these 3?
Could you share where your references for this?

File: sktime/networks/resnet/_resnet_torch.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            "relu": "torch.nn.ReLU",
+            "tanh": "torch.nn.Tanh",
+            "sigmoid": "torch.nn.Sigmoid",
+        }
+        if activation_hidden not in _supported:
```
Comment (RecreationalMath) [inline sktime/networks/resnet/_resnet_torch.py]: `activation_hidden` should be used in each block not just at the end.

**Review by RecreationalMath — CHANGES_REQUESTED** [review]
I've left some comments for you to look over.
I'd also encourage you to go through the comments on other open/merged PRs for the umbrella migration issue as a lot of the mistakes have already been flagged and rectified there.
To be frank, this smells a lot like AI-generated and very little self-coding.

**Comment by RecreationalMath** [comment]
Please refer to comments on #9298 , that should help here.

**Comment by DCchoudhury15** [comment]
Hey @RecreationalMath, pushed a second commit addressing everything you pointed out plus a few things I caught along the way.

**Your review items:** reordered `self.X = X` assignments in both classifier and regressor to match `__init__` parameter order, and removed the `_safe_import` caching pattern in `_init_weights` now calls `_safe_import` directly each time, matching how the RNN network handles it.

**Additional changes:** added the `"python_version": ">=3.10,<3.15"` tag to all three files (was missing, needs to match `pyproject.toml`); cleaned up docstrings by removing filler text and fixing parameter ordering to match `__init__` signatures; added `kernel_sizes` length validation to the classifier and regressor constructors so it fails fast instead of surfacing deep inside `_build_network`  got this from your review on srupat's TapNet PR (#9412) about catching bad parameters early; added `# model specific` and `# base classifier specific` / `# base regressor specific` separator comments in `__init__` parameter lists to match the RNN and TapNet convention; expanded classifier test params to 5 configurations covering different activation/criterion combos (`CrossEntropyLoss`, `BCELoss`, `BCEWithLogitsLoss`, `NLLLoss`) instead of just the 2 basic ones.

**Things I'd like your input on:** (1) I added a `get_test_params` classmethod to `ResNetNetworkTorch` for standalone testing, but the RNN network doesn't have one  it doesn't cause issues but let me know if you'd prefer I remove it for consistency; (2) the TF ResNet classifier defaults to `activation="sigmoid"`, but I went with `activation=None` for PyTorch because `CrossEntropyLoss` (our default criterion) expects raw logits and applying sigmoid would be incorrect  wanted to confirm this reasoning makes sense.

**Tests:** full suite passing before push - 378 classifier, 101 regressor, pre-commit clean.

**Comment by DCchoudhury15** [comment]
@RecreationalMath gentle ping
